packages/vector_graphics_compiler/lib/src/svg/masking_optimizer.dart

```diff
@@ -0,0 +1,367 @@
+        {
+          newCommands.add(LineToCommand(points[index++], points[index++]));
+        }
+        break;
+      case path_ops.PathVerb.cubicTo:
+        {
+          newCommands.add(CubicToCommand(
+            points[index++],
+            points[index++],
+            points[index++],
+            points[index++],
+            points[index++],
+            points[index++],
+          ));
+        }
+        break;
+      case path_ops.PathVerb.close:
+        newCommands.add(const CloseCommand());
+        break;
+    }
+  }
+
+  Path newPath = Path(commands: newCommands);
+
+  if (path.fillType == path_ops.FillType.evenOdd) {
+    newPath = Path(commands: newCommands, fillType: PathFillType.evenOdd);
+  }
+
+  return newPath;
+}
+
+/// Gets the single child recursively,
+/// returns null if there are 0 children or more than 1
+ResolvedPathNode? getSingleChild(Node node) {
+  if (node.runtimeType.toString() == 'ResolvedPathNode') {
+    return node as ResolvedPathNode;
+  } else if (node.runtimeType.toString() == 'ParentNode') {
+    if ((node as ParentNode).children.length == 1) {
+      return getSingleChild(node.children.single);
+    } else {
+      return null;
+    }
+  } else {
+    return null;
+  }
+}
+
+/// Traverses all children of a given node
+List<T> queryChildren<T extends Node>(Node node) {
+  final List<T> children = <T>[];
+  void visitor(Node child) {
+    if (child is T) {
+      children.add(child);
+    }
+    child.visitChildren(visitor);
+  }
+
+  node.visitChildren(visitor);
+  return children;
+}
+
+/// Checks if one shape is inside another using rectagle bounds
+bool isInside(Rect outerShape, Rect innerShape) {
+  return ((innerShape.left > outerShape.left) &&
+      (innerShape.right < outerShape.right) &&
+      (innerShape.top > outerShape.top) &&
+      (innerShape.bottom < outerShape.bottom));
+}
+
+/// Parses SVG string to node tree and runs it through the resolver
+Future<Node> parseAndResolve(String source) async {
+  final Node node = await parseToNodeTree(source);
+  final ResolvingVisitor visitor = ResolvingVisitor();
+  return node.accept(visitor, AffineMatrix.identity);
+}
+
+/// Simplifies masking operations into PathNodes
+class MaskingOptimizer extends Visitor<_Result, Node>
+    with ErrorOnUnResolvedNode<_Result, Node> {
+  ///List of masks to add.
+  List<ResolvedPathNode> masksToApply = [];
+
+  /// Applies visitor to given node
+  Node apply(Node node) {
+    final Node newNode = node.accept(this, null).node;
+    return newNode;
+  }
+
+  /// Applies mask to a path node, and returns resulting path node.
+  ResolvedPathNode applyMask(Node child, ResolvedPathNode maskPathNode) {
+    final ResolvedPathNode pathNode = child as ResolvedPathNode;
+    final path_ops.Path maskPathOpsPath = toPathOpsPath(maskPathNode.path);
+    maskPathOpsPath.applyOp(
+        toPathOpsPath(pathNode.path), path_ops.PathOp.intersect);
+    final Path newPath = toVectorGraphicsPath(maskPathOpsPath);
```

Comment:
This API got refactored, now you'll want something like

```
final path_ops.Path intersection = maskPathOpsPath.applyOp(....)
```

Also, you'll need to make sure to dispose the `path_ops.Path` objects, otherwise you're leaking memory. That means you'll need to have a variable for the result of `toPathOpsPath` so you can call `.dispose()` on it.

Reply:
fixed.